sr_multi_moveit/sr_multi_moveit_config/scripts/generate_robot_srdf.py

```diff
@@ -394,50 +407,94 @@ def parse_arm_groups(self, manipulator_id, manipulator):
+                new_group_state.setAttribute("group", move_group_name)
+                new_group_state.setAttribute("name", move_group_state_name)
+                for joint_name, joint_angle in move_group_state.items():
+                    new_group_state.appendChild(xml.dom.minidom.Document().createElement(
+                        f'joint name="{joint_name}" value="{joint_angle}"'))
+                new_group_state.writexml(self.new_robot_srdf, indent="  ", addindent="  ", newl="\n")
+
+    # Creates a new move group state, resolving any values inherited from other move groups
+    def create_move_group_state(self, group_name, group_state_name):
+        combined_move_group_state = self._combined_move_group_states[group_name][group_state_name]
+        if "joint_angles" not in combined_move_group_state:
+            combined_move_group_state["joint_angles"] = {}
+        # If the new move group state inherits from any others
+        if "inherit_from" in combined_move_group_state:
+            for ancestor in combined_move_group_state["inherit_from"]:
+                ancestor_group = ancestor["move_group"]
+                ancestor_group_state = ancestor["move_group_state"]
+                # If the ancestor group state is not defined yet
+                if (ancestor_group not in self._move_group_states or
+                        ancestor_group_state not in self._move_group_states[ancestor_group]):
+                    # If the ancestor group state is not even defined in the combined move group states
+                    if (ancestor_group not in self._combined_move_group_states or
+                            ancestor_group_state not in self._combined_move_group_states[ancestor_group]):
+                        rospy.logwarn(f'Unable to create move group "{group_name}" state "{group_state_name}", as it '
+                                      f'inherits from move group "{ancestor_group}" state "{ancestor_group_state}", '
+                                      f'which is not defined.')
+                        return False
+                    else:
+                        # Try to create the ancestor group state, return if failed
+                        if not self.create_move_group_state(ancestor_group, ancestor_group_state):
+                            return False
+                # The ancestor group state either already existed or has been created
+                ancestor_move_group_state = self._move_group_states[ancestor_group][ancestor_group_state]
+                for joint, value in ancestor_move_group_state.items():
+                    if joint not in combined_move_group_state["joint_angles"]:
+                        combined_move_group_state["joint_angles"][joint] = value
+        if group_name not in self._move_group_states:
+            self._move_group_states[group_name] = {}
+        self._move_group_states[group_name][group_state_name] = combined_move_group_state["joint_angles"]
```

Comment:
I'm lacking a bit of context about the whole information flow of srdfs and yamls into these data structures.
Not sure what do we do with `_move_group_states` and `_new_move_group_states`.
Is there a document describing a bit this flow? If not, would it be a good time to write an initial draft, if that's clear in your head right now?

Reply:
Added some documentation and renamed some variables to make things clearer in https://github.com/shadow-robot/sr_interface/pull/768/commits/c17efa1cd76b3ae46e55204f9b9d8b4b765f4c4c.